pkg/channelwatcher/channel_watcher.go

```diff
@@ -57,3 +58,53 @@ func New(mgr manager.Manager, logger *zap.Logger, watchHandler WatchHandlerFunc)
 	}
 	return nil
 }
+
+// WatchHandlerFunc is called whenever an add, update, delete or generic event is triggered on a channel
+type WatchHandlerFunc func(context.Context, client.Client, types.NamespacedName) error
+
+// ShouldWatchFunc is called while returning list of channels.
+// Channels are included in the list if the return value is true.
+type ShouldWatchFunc func(ch *v1alpha1.Channel) bool
+
+// UpdateConfigWatchHandler is a special handler that
+// 1. Lists the channels for which shouldWatch returns true
+// 2. Creates a multi-channel-fanout-config
+// 3. Calls the updateConfig func with the new multi-channel-fanout-config
+// This is used by dispatchers or receivers to update their configs by watching channels
+func UpdateConfigWatchHandler(updateConfig swappable.UpdateConfig, shouldWatch ShouldWatchFunc) WatchHandlerFunc {
+	return func(ctx context.Context, c client.Client, _ types.NamespacedName) error {
+		channels, err := listAllChannels(ctx, c, shouldWatch)
+		if err != nil {
+			logging.FromContext(ctx).Info("Unable to list channels", zap.Error(err))
+			return err
+		}
+		config := multichannelfanout.NewConfigFromChannels(channels)
+		return updateConfig(config)
+	}
+}
+
+// listAllChannels queries client and gets list of all channels for which shouldWatch returns true.
+func listAllChannels(ctx context.Context, c client.Client, shouldWatch ShouldWatchFunc) ([]v1alpha1.Channel, error) {
+	channels := make([]v1alpha1.Channel, 0)
+	for {
+		cl := &v1alpha1.ChannelList{}
+		opts := &client.ListOptions{
+			// Set Raw because if we need to get more than one page, then we will put the continue token
+			// into opts.Raw.Continue.
+			Raw: &metav1.ListOptions{},
```

Comment:
Based on https://github.com/knative/eventing/pull/1044#discussion_r275572557, we don't need to do the list chunking.